hashicorp/consul
# Rename hcp-metrics-collector to consul-telemetry-collector

### Description

We originally named the collector service name as hcp-metrics-collector. Rename the service name to follow the intended project name.

Also renamed the `bind_socket_dir` config key to remove the `hcp` reference. The collector can be used without HCP so it made sense to generalize the name

### Testing & Reproduction steps
- Unit tests have been modified to ensure everything passes.
- Tested locally as follows: 
1. Run consul agent in dev mode: `bin/consul agent -dev`
2. Register services:
  - web service: `consul services register web.json`
    ```json
    {
    "service": {
        "name": "web",
        "port": 9191,
        "connect": {
            "sidecar_service": {
                "proxy": {
                    "config": {
                        "envoy_telemetry_collector_bind_socket_dir": "/Users/ashvitha.sridharan/tmp/"
                    }
                }
            }
        }
    }
}```

   - collector service: `consul services register collector.json`
  ```json
{    "service":{
        "name": "consul-telemetry-collector",
        "port": 9090,
        "connect": {
            "sidecar_service": {
            }
        }
    }
}
```
2. Run sidecars
- `./bin/consul connect envoy -sidecar-for web -- -l debug`
- `./bin/consul connect envoy -admin-bind localhost:19001 -sidecar-for consul-telemetry-collector -- -l debug`

3. Run a mock metric handler (grpc service) and see metrics flow:
4. <img width="790" alt="Screenshot 2023-05-12 at 3 40 46 PM" src="https://github.com/hashicorp/consul/assets/23405655/39cb7f34-cdef-4fa4-b03e-8ae69a91f307">

### Links



### PR Checklist
